### PR TITLE
fix #154

### LINKED ISSUE #154: Make debouncing "save preferences" optional
I implemented this awhile back, now I'd like the option to turn it off. Pretty simple, I'll have a PR soon.

### PR DESCRIPTION
**Pull request by lacymorrow**
Adds a `config.debounce` options, also moves `css` and `dataStore` into `config` and adds a public getter

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,12 +18,30 @@ class ElectronPreferences extends EventEmitter2 {
...
 				devTools: false,
 			},
 		});
 
+		this.options = options;
+
+		// Legacy: Set config values
+		if (options.css && !options.config.css) {
+
+			this.options.config.css = options.css;
```
Comment (pvrobays) [inline index.js:35]: Thanks for thinking of backwards compatibility.
In this case I would add a deprecation warning, indicating `options.css` will be removed in future versions.
```suggestion
			console.warn("DEPRECATED: css option has been deprecated and will be removed in a future version. It now lives under config.css.");
			this.options.config.css = options.css;
```

File: index.js [hunk 2]
```diff
@@ -18,12 +18,30 @@ class ElectronPreferences extends EventEmitter2 {
...
+
+		}
+
+		if (options.dataStore && !options.config.dataStore) {
+
+			this.options.config.dataStore = options.dataStore;
```
Comment (pvrobays) [inline index.js:41]: Same thing here, I'd add a warning message:
```suggestion
			console.warn("DEPRECATED: dataStore option has been deprecated and will be removed in a future version. It now lives under config.dataStore.");
			this.options.config.dataStore = options.dataStore;
```

File: index.js [hunk 3]
```diff
@@ -336,9 +364,10 @@ class ElectronPreferences extends EventEmitter2 {
 		this.prefsWindow.webContents.on('dom-ready', async () => {
 
 			// Load custom css file
-			if (this.options.css) {
+			const cssFile = this.options.config.css;
```
Comment (pvrobays) [inline index.js:367]: Can be simplified:
```suggestion
			const cssFile = this.config.css;
```

**Comment by pvrobays** [comment]
@lacymorrow LGTM. I'll merge when conflicts are resolved.
Would you mind adding the debounce option to the CHANGELOG.md?

**Comment by lacymorrow** [comment]
merged and added

**Comment by pvrobays** [comment]
@lacymorrow Sorry, again conflicts

**Comment by lacymorrow** [comment]
I'm not 100% that `config.css` and `config.datastore` are working 100%, I'm busy this week but will try to test next week

**Comment by lacymorrow** [comment]
I really want this feature, and have been using it in my branch for awhile, but I realized only certain components need to be debounced, rather than the whole save process.

**Comment by pvrobays** [comment]
I agree with the feature. For some devs it may be useful to debounce the saving process as the saving callback might trigger high resource intensive calls on their side.
What do you mean with 'only certain components need to be debounced'? Do you have an example of what you mean with 'components'?

Once conflicts are resolved, this PR can get merged.

**Comment by lacymorrow** [comment]
So you want to debounce things that change rapidly. The downside is that instead of your changes being displayed immediately, they are debounced a bit to avoid a thousand DOM updates per second. 

A radio group/checkbox have no reason to be debounced since it takes time to move your mouse between items

**Comment by pvrobays** [comment]
Ok, now I get what you mean by component.
IMO I would just go ahead and debounce the save calls to make the implementation less complex. Also, I think you can adapt radio buttons with the arrow keys, leaking you could easily trigger hundredth of calls per second, even on a radio button.

Let me know what you think/want to implement. I don't have a strong opinion on this.

**Comment by lacymorrow** [comment]
fair enough, fixed the merge conflicts, I'll revisit this soon

**Comment by pvrobays** [comment]
@lacymorrow Sorry, this got lost in my notifications. Merged!

**Pull request merged**